onnxruntime/core/optimizer/transpose_optimizer/api.h

```diff
@@ -430,60 +473,72 @@ constexpr int64_t kMaxSupportedOpset = 15;
 bool Optimize(api::GraphRef& graph, bool allow_extended_ops);
 
 /* Layout Transformation Tools
- * These methods change the channel ordering of layout sensitive ops (like Conv). They work by replacing the op with
- * an op taking the new ordering (usually a com.microsoft op) and surrounding it with transposes to keep the model
- * correct. Then (if any replacements were made), they call Optimize on the graph.
- *
- * To use, create a LayoutHandler function for each op. Then provide the handler map to ChannelLastToChannelFirst or
- * ChannelFirstToChannelLast. If the parameters of these convenience methods are too restrictive, you can swap the
- * ops for reordered versions yourself and call Optimize manually.
+ * These methods help change the channel ordering of layout sensitive ops (like Conv). ONNX currently only supports
+ * channel first ordering for ops, so this requires changing the op type and domain to a contrib op supporting
+ * the new ordering. The existence of a robust transpose optimizer means that we can freely add transpose ops during
+ * conversion and then call Optimize to remove as many as possible. To change the channel ordering of some/all ops
+ * in a model, a user of this tool should do the following:
+ * 
+ * 1. Iterate over the graph nodes and identify nodes for convert. For each one:
+ *    a. Change the op type and domain (and possibly attributes) to the op/contrib op with the desired ordering.
+ *    b. The model is currently invalid since the input tensors are in the original ordering and all consumers
+ *       expect the original ordering. Use WrapTransposesAroundNode helper to insert transposes around the
+ *       inputs/outputs of the op to correct this.
+ * 2. The model is now correct but has many unnecessary Transpose ops. Call Optimize on the graph.
+ * 
+ * After step 1, the Transpose ops will wrap converted ops in a similar manner to q/dq ops in quantization.
+ * The perm attributes essentially encode the information about which ops are being reordered.
  */
 
 /// <summary>
-/// Returned by a LayoutHandler function. Indicates whether reordering should occur and what op type to use.
-/// Modifications to attributes can be performed by the LayoutHandler function itself, since all attribute updates
-/// will persist.
+/// Inserts transposes around op inputs/outputs. Alternatively transposes initializers or uses existing Transpose
+/// nodes if possible. Populates shape information on affected node inputs/outputs to reflect the change.
+/// 
+/// Ex:
+///   * -> Conv -> **
+///   becomes
+///   * -> Transpose -> Conv -> Transpose -> **
+///   Model is now invalid until Conv op type is changed. Conv inputs/outputs have new shape. Shapes of * and **
+///   are unchanged.
```

Comment:
nit: this doesn't seem to match with above where it says step 1 is to change the op type and domain (rendering the graph invalid) and step 2 is to insert the Transposes to fix. 

would have expected a line in the middle with * --> Conv.NHWC -> ** or something like that